src/RadhydroPulseMGint/test_radhydro_pulse_MG_int.cpp

```diff
@@ -0,0 +1,551 @@
+constexpr double T1 = 2.0e7; // K (temperature)
+constexpr double rho0 = 1.2; // g cm^-3 (matter density)
+constexpr double a_rad = C::a_rad;
+constexpr double c = C::c_light; // speed of light (cgs)
+constexpr double chat = c;
+constexpr double width = 24.0; // cm, width of the pulse
+constexpr double erad_floor = a_rad * T0 * T0 * T0 * T0 * 1.0e-10;
+constexpr double mu = 2.33 * C::m_u;
+constexpr double h_planck = C::hplanck;
+constexpr double k_B = C::k_B;
+
+// static diffusion: (for single group) tau = 2e3, beta = 3e-5, beta tau = 6e-2
+constexpr double kappa0 = 180.;	    // cm^2 g^-1
+constexpr double v0_adv = 1.0e6;    // advecting pulse
+constexpr double max_time = 4.8e-6; // max_time = 2.0 * width / v1;
+
+// dynamic diffusion: tau = 2e4, beta = 3e-3, beta tau = 60
+// constexpr double kappa0 = 1000.; // cm^2 g^-1
+// constexpr double v0_adv = 1.0e8;    // advecting pulse
+// constexpr double max_time = 1.2e-4; // max_time = 2.0 * width / v1;
+
+constexpr double T_ref = T0;
+constexpr double nu_ref = 1.0e18;			     // Hz
+constexpr double coeff_ = h_planck * nu_ref / (k_B * T_ref); // = 4.799243073 = 1 / 0.2083661912
+
+template <> struct quokka::EOS_Traits<MGintProblem> {
+	static constexpr double mean_molecular_weight = mu;
+	static constexpr double boltzmann_constant = k_B;
+	static constexpr double gamma = 5. / 3.;
+};
+template <> struct quokka::EOS_Traits<GreyProblem> {
+	static constexpr double mean_molecular_weight = mu;
+	static constexpr double boltzmann_constant = k_B;
+	static constexpr double gamma = 5. / 3.;
+};
+
+template <> struct Physics_Traits<MGintProblem> {
+	// cell-centred
+	static constexpr bool is_hydro_enabled = true;
+	static constexpr int numMassScalars = 0;		     // number of mass scalars
+	static constexpr int numPassiveScalars = numMassScalars + 0; // number of passive scalars
+	static constexpr bool is_radiation_enabled = true;
+	// face-centred
+	static constexpr bool is_mhd_enabled = false;
+	static constexpr int nGroups = n_groups_;
+};
+template <> struct Physics_Traits<GreyProblem> {
+	// cell-centred
+	static constexpr bool is_hydro_enabled = true;
+	static constexpr int numMassScalars = 0;		     // number of mass scalars
+	static constexpr int numPassiveScalars = numMassScalars + 0; // number of passive scalars
+	static constexpr bool is_radiation_enabled = true;
+	// face-centred
+	static constexpr bool is_mhd_enabled = false;
+	static constexpr int nGroups = 1;
+};
+
+template <> struct RadSystem_Traits<MGintProblem> {
+	static constexpr double c_light = c;
+	static constexpr double c_hat = chat;
+	static constexpr double radiation_constant = a_rad;
+	static constexpr double Erad_floor = erad_floor;
+	static constexpr bool compute_v_over_c_terms = true;
+	static constexpr double energy_unit = h_planck;
+	static constexpr amrex::GpuArray<double, n_groups_ + 1> radBoundaries = rad_boundaries_;
+	static constexpr int beta_order = 1;
+	static constexpr int opacity_model = 1;
```

Comment:
Can we provide an enum here in place of an integer value for opacity model, in order to make the code more readable?

Reply:
Done.